src/System.Net.Sockets/tests/FunctionalTests/TimeoutTest.cs

```diff
@@ -68,22 +66,20 @@ public void ReceiveTimesOut_Throws(bool forceNonBlocking)
 
                 acceptedSocket.ForceNonBlocking(forceNonBlocking);
 
-                DateTime start = default(DateTime);
+                long start = Environment.TickCount64;
 
                 SocketException sockEx = Assert.Throws<SocketException>(() =>
                 {
-                    start = DateTime.UtcNow;
                     acceptedSocket.Receive(new byte[1]);
                 });
 
-                double elapsed = (DateTime.UtcNow - start).TotalMilliseconds;
-
+                long elapsed = Environment.TickCount64 - start;
                 Assert.Equal(SocketError.TimedOut, sockEx.SocketErrorCode);
                 Assert.True(acceptedSocket.Connected);
 
                 // Try to ensure that the elapsed timeout is reasonably correct
                 // Sometimes test machines run slowly
-                Assert.InRange(elapsed, Timeout * 0.75, Timeout * 2);
+                Assert.InRange(elapsed, Timeout * 0.75, Timeout * 1.9999);
```

Comment:
The unit of measure for `elapsed` has been changed from milliseconds to ticks, but it looks like `Timeout` is unmodified and presumably still using milliseconds. Is this assertion still valid?

Reply:
Environment.TickCount is milliseconds.

Reply:
TIL thanks

Reply:
Yeah, it's not intuitive.